docs/source/datasets.rst

```diff
@@ -3,6 +3,61 @@ torchtext.datasets
 
 .. currentmodule:: torchtext.datasets
 
+
+.. _datapipes_warnings:
+
+.. warning::
+
+    The datasets supported by torchtext are datapipes from the `torchdata
+    project <https://pytorch.org/data/beta/index.html>`_, which is still in Beta
+    status. This means that the API is subject to change without deprecation
+    cycles. In particular, we expect a lot of the current idioms to change with
+    the eventual release of ``DataLoaderV2`` from ``torchdata``.
+
+    Here are a few recommendations regarding the use of datapipes:
+
+    - For shuffling the datapipe, do that in the DataLoader: ``DataLoader(dp, shuffle=True)``.
+      You do not need to call ``dp.shuffle()``, because ``torchtext`` has
+      already done that for you. Note however that the datapipe won't be
+      shuffled unless you explicitly pass ``shuffle=True`` to the DataLoader.
+
+    - When using multi-processing (``num_workers=N``), use the builtin ``worker_init_fn``::
+
+            from torch.utils.data.backward_compatibility import worker_init_fn
+            DataLoader(dp, num_workers=4, worker_init_fn=worker_init_fn, drop_last=True)
+
+      This will ensure that data isn't duplicated across workers.
+
+    - We also recommend using ``drop_last=True``. Without this, the batch sizes
+      at the end of an epoch may be very small in some cases (smaller than with
+      other map-style datasets). This might affect accuracy greatly especially
+      when batch-norm is used. ``drop_last=True`` ensures that all batch sizes
+      are equal.
+
+    - Distributed training with ``DistributedDataParallel`` is not yet entirely
+      stable / supported, and we don't recommend it at this point. It will be
+      better supported in DataLoaderV2. If you still wish to use DDP, make sure
+      that:
+
+      - All workers (DDP workers *and* DataLoader workers) see a different part
+        of the data. The datasets are already wrapped inside  `ShardingFilter
+        <https://pytorch.org/data/main/generated/torchdata.datapipes.iter.ShardingFilter.html>`_
+        and you may need to call ``dp.apply_sharing(num_shards, shard_id)`` in order to shard the 
+        data across ranks (DDP workers) and DataLoader workers. One way to do this
+        is to create ``worker_init_fn`` that calls ``apply_sharding`` with appropriate 
+        number of shards (DDP workers * DataLoader workers) and shard id (inferred through rank
+        and worker ID of corresponding DataLoader withing rank). Note however, that this assumes
+        equal number of DataLoader workers for all the ranks. 
```

Comment:
@NicolasHug, @VitalyFedyunin , @ejguan : I made the description rather high level while leaving implementation details to the users. My thinking here is to make the user aware of pitfalls without necessarily providing specific recommendation on implementing specific solution.